Algorithms_and_Data_Structures/Arrays/Single_number.py

```
@@ -0,0 +1,27 @@
+        once ^= num
+        # common_mask removes bits appearing three times
+        common_mask = ~(once & twice)
+        once &= common_mask
+        twice &= common_mask
+
+    return once
+
+if __name__ == "__main__":
+    nums = []
+    n = int(input("Enter number of elements in array: "))
+    input_array(nums, n)
```

Ruff (0.12.2)

Undefined name `input_array`

(F821)